node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeagent/NodeAgentImpl.java

```diff
@@ -356,6 +359,27 @@ public void suspend() {
         return Optional.of(existingContainer);
     }
 
+    private void updateContainerIfNeeded(NodeAgentContext context, Container existingContainer) {
+        double cpuCap = context.node().getOwner()
+                .map(NodeSpec.Owner::asApplicationId)
+                .map(appId -> containerCpuCap.with(FetchVector.Dimension.APPLICATION_ID, appId.serializedForm()))
+                .orElse(containerCpuCap)
+                .value();
+
+        ContainerResources wantedContainerResources = ContainerResources.from(
+                cpuCap, context.node().getMinCpuCores(), context.node().getMinMainMemoryAvailableGb());
+
+        if (wantedContainerResources.equals(existingContainer.resources)) return;
+        context.log(logger, "Container should be running with different resource allocation, wanted: %s, current: %s",
+                wantedContainerResources, existingContainer.resources);
+
+        if (context.node().getState() == Node.State.active) {
```

Comment:
This method is always called when active?

Reply:
True, moved this check to `orchestratorSuspendNode()` in case we sometime add the update to any other state.